src/sentry/tasks/integrations/migrate_issues.py

```diff
@@ -0,0 +1,109 @@
+from django.db import IntegrityError, transaction
+
+from sentry.models import (
+    ExternalIssue,
+    Group,
+    GroupLink,
+    GroupMeta,
+    Integration,
+    OrganizationIntegration,
+    Project,
+)
+from sentry.plugins.base import plugins
+from sentry.tasks.base import instrumented_task, retry
+from sentry.tasks.integrations import logger
+
+
+@instrumented_task(
+    name="sentry.tasks.integrations.migrate_issues",
+    queue="integrations",
+    default_retry_delay=60 * 5,
+    max_retries=5,
+)
+@retry(exclude=(Integration.DoesNotExist))
+def migrate_issues(integration_id: int, organization_id: int) -> None:
+    from sentry_plugins.jira.plugin import JiraPlugin
+
+    integration = Integration.objects.get(id=integration_id)
+    for project in Project.objects.filter(organization_id=organization_id):
+        plugin = None
+        for p in plugins.for_project(project):
+            if isinstance(p, JiraPlugin) and p.is_configured(None, project):
+                plugin = p
+                break
+
+        if not plugin:
+            continue
+
+        is_different_jira_instance = plugin.get_option("instance_url", project).rstrip(
+            "/"
+        ) != integration.metadata.get("base_url").rstrip("/")
+        if is_different_jira_instance:
+            continue
+        groups = Group.objects.filter(project=project.id)
+        # this seems expensive but I don't think there are other options
+        plugin_issues = GroupMeta.objects.filter(
+            key=f"{plugin.slug}:tid", group__id__in=[group.id for group in groups]
+        )
```

Comment:
```suggestion
        plugin_issues = GroupMeta.objects.filter(
            key=f"{plugin.slug}:tid", group__project__id=project.id
        )
```

This could be *really* big. GroupMeta has a foreign key to group, and group to project. You should be able to use joins instead of loading all the groups for a project into memory.